### PR TITLE
Document some exposed functions

### PR DESCRIPTION
**Pull request by jhrcek**

### CODE REVIEW

File: src/HieDb/Create.hs [hunk 1]
```diff
@@ -135,7 +143,7 @@ initConn (getConn -> conn) = do
 
 addArr :: HieDb -> A.Array TypeIndex HieTypeFlat -> IO (A.Array TypeIndex (Maybe Int64))
 addArr (getConn -> conn) arr = do
-  fmap (A.listArray (A.bounds arr)) $ forM (A.elems arr) $ \case
+  forM arr $ \case
```
Comment (jhrcek) [inline src/HieDb/Create.hs:146]: Array is Traversable, no need to convert it to list and then back to array.

File: src/HieDb/Create.hs [hunk 2]
```diff
@@ -154,14 +162,17 @@ addTypeRefs :: HieDb -> FilePath -> HieFile -> A.Array TypeIndex (Maybe Int64) -
 addTypeRefs db path hf ixs = mapM_ addTypesFromAst asts
   where
     arr = hie_types hf
-    asts = M.elems $ getAsts $ hie_asts hf
+    asts = getAsts $ hie_asts hf
```
Comment (jhrcek) [inline src/HieDb/Create.hs:165]: Map is traversable too, so avoid unnecessary conversion to list.

File: src/HieDb/Query.hs [hunk 3]
```diff
@@ -113,31 +120,34 @@ searchDef conn cs
...
+In case the target is given by ModuleName (and optionally UnitId) it is first resolved
+from HieDb, which can lead to error if given file is not indexed/Module name is ambiguous.
+-}
 withTarget
   :: HieDb
-  -> Either FilePath (ModuleName, Maybe UnitId)
+  -> HieTarget
   -> (HieFile -> a)
   -> IO (Either HieDbErr a)
-withTarget conn (Left x') f = do
```
Comment (jhrcek) [inline src/HieDb/Query.hs:121]: Just factored out the logic common to both branches.

File: src/HieDb/Types.hs [hunk 4]
```diff
@@ -49,11 +49,12 @@ setHieTrace (getConn -> conn) = setTrace conn
...
   { modInfoName :: ModuleName
-  , modInfoUnit :: UnitId
-  , modInfoIsBoot :: Bool
-  , modInfoSrcFile :: Maybe FilePath
-  , modInfoIsReal :: Bool -- Does this represent an actual file or a cached dep?
-  , modInfoTime :: UTCTime
+  , modInfoUnit :: UnitId -- ^ Identifies the package this module is part of
+  , modInfoIsBoot :: Bool -- ^ True, when this ModuleInfo was created by indexing .hie-boot file;
+                          -- False when it was created from .hie file
+  , modInfoSrcFile :: Maybe FilePath -- ^ Tha path to the haskell source file, from which the .hie file was created
```
Comment (wz1000) [inline src/HieDb/Types.hs:55]:
```suggestion
  , modInfoSrcFile :: Maybe FilePath -- ^ The path to the haskell source file, from which the .hie file was created
```

**Comment by wz1000** [comment]
Very nice.

**Pull request merged**